lexnlp/ml/tests/test_model_io.py

```
@@ -0,0 +1,324 @@
+        assert _load_legacy(path) == {"jl": True}
+
+    def test_unknown_suffix_falls_back_to_pickle(self, tmp_path: Path) -> None:
+        """An unknown extension should still load if the content is a valid pickle."""
+        path = tmp_path / "model.bin"
+        _write_pickle(path, "hello")
+        assert _load_legacy(path) == "hello"
+
+    def test_pickle_file_invalid_raises(self, tmp_path: Path) -> None:
+        path = tmp_path / "corrupt.pickle"
+        path.write_bytes(b"not a pickle")
+        with pytest.raises(Exception):
```

Ruff (0.15.10)

[warning] Do not assert blind exception: `Exception`

(B017)